Modules/_functoolsmodule.c

```diff
@@ -1257,6 +1257,15 @@ lru_cache_cache_info(lru_cache_object *self, PyObject *unused)
                                  PyDict_GET_SIZE(self->cache));
 }
 
+static PyObject *
+lru_cache_cache_parameters(lru_cache_object *self)
+{
+    PyObject *cache_parameters =  PyDict_New();
+    PyDict_SetItemString(cache_parameters, "maxsize", PyLong_FromSsize_t(self->maxsize));
+    PyDict_SetItemString(cache_parameters, "typed", self->typed == 0 ? Py_False : Py_True);
```

Comment:
1.  Drop the test for zero, putting the positive case first: ``self->typed ? Py_True : Py_False)``
2. Test the result of *PyDict_SetItemString()* to see if it failed.